packages/core/src/browser/view-container.ts

```diff
@@ -750,13 +750,17 @@ export class ViewContainerPart extends BaseWidget {
         this.collapsedEmitter.fire(collapsed);
     }
 
-    setHidden(hidden: boolean): void {
+    setHidden(hidden: boolean, refreshToolbar?: boolean): void {
         if (!this.canHide) {
             return;
         }
         super.setHidden(hidden);
         if (!this.isHidden) {
             this.collapsed = false;
+            if (refreshToolbar) {
```

Comment:
1. maybe move refresh toolbar to its own method and call it separately from `setHidden`?
2. could refresh of toolbar be done without calling hide and then show? it seems a little bit strange. does it cause flickering?

Reply:
@amiramw 
1) The underlying logic here is if you are not hiding widget - refresh toolbar if mouse is hovering over it. So based on that
I combined both actions to one method invocation - as refreshtoolbar is dependent on hiding or showing widget and thought adding an argument to setHidden is cleaner than adding another function call.
2) I agree as I stated above in explanation to Vince that this solution is not ideal but based on existing constraints and not wanting to refactor entire logic there - I thought this is a good enough solution for this requirement.
"_4. I agree that ideally setHidden and refreshToolbar are not supposed to be connected and one doesn't imply the other - but I thought that the commonality outweighs the difference._"

Obviously if you think otherwise let me know and I'll change.
Thanks